tests/scancode/test_resource.py

```diff
@@ -1168,4 +1175,41 @@ def test_VirtualCodebase_can_be_created_from_dict(self):
 
         results = sorted(r.name for r in codebase.walk())
         expected = ['bar.svg', 'han']
-        assert expected == results
\ No newline at end of file
+        assert expected == results
+
+    def test_VirtualCodebase_create_from_scan_with_no_root_and_missing_parents(self):
+        test_file = self.get_test_loc('resource/virtual_codebase/samples-only-findings.json')
+        codebase = VirtualCodebase(test_file)
+        results = sorted(r.name for r in codebase.walk())
+        expected = [
```

Comment:
This expectation is not really easy to make sense off... It would be best to either use a file to expect the whole Resource tree